test_thurstone_scaling failures with SciPy 1.14

Since upgrading to SciPy 1.14 in Debian, we are seeing `test_thurstone_scaling` failing on some architectures (ARM, POWER and RISC-V, among others).

```
>           assert np.isclose(next_stim_1, expected_stim_1)
E           assert False
E            +  where False = <function isclose at 0xffff9d268ef0>(1.0, 0.6)
E            +    where <function isclose at 0xffff9d268ef0> = np.isclose
```
 
The new failure occurs when `trial_idx` is 21, so I hoped the following change would resolve it:

```
-        if trial_idx in (6, 20):
+        if trial_idx in (6, 20, 21):
```

However, at the time of the failure, the values are:

```
next_stim_1:  1.0
expected_stim_1:  0.6
next_stim_2:  0.6000000000000001
expected_stim_2:  1.0
```

And  the following change was also needed:

```
-                expected_stim_1 == next_stim_2 and
-                expected_stim_2 == next_stim_1
+                np.isclose(expected_stim_1, next_stim_2) and
+                np.isclose(expected_stim_2, next_stim_1)
```

--- a/questplus/tests/test_qp.py
+++ b/questplus/tests/test_qp.py
@@ -435,11 +435,11 @@ def test_thurstone_scaling():
         next_stim_1 =  qp.next_stim['physical_magnitude_stim_1']
         next_stim_2 =  qp.next_stim['physical_magnitude_stim_2']
 
-        if trial_idx in (6, 20):
+        if trial_idx in (6, 20, 21):
             # Rounding errors make the algorithm behave differently on different platforms.
             if (
-                expected_stim_1 == next_stim_2 and
-                expected_stim_2 == next_stim_1
+                np.isclose(expected_stim_1, next_stim_2) and
+                np.isclose(expected_stim_2, next_stim_1)
             ):
                 expected_stim_1, expected_stim_2 = expected_stim_2, expected_stim_1
                 response = 'First' if response == 'Second' else 'Second'

